skimage/restoration/_denoise.py

```diff
@@ -470,3 +470,94 @@ def denoise_wavelet(img, sigma=None, wavelet='db1', mode='soft'):
 
     clip_range = (-1, 1) if img.min() < 0 else (0, 1)
     return np.clip(out, *clip_range)
+
+
+def _sigma_est_dwt(detail_coeffs, distribution='Gaussian'):
+    """
+    Calculation of the robust median estimator of the noise standard
+    deviation.
+
+    Parameters
+    ----------
+    detail_coeffs : ndarray
+        The detail coefficients corresponding to the discrete wavelet
+        transform of an image.
+    distribution : str
+        The underlying noise distribution.
+
+    Returns
+    -------
+    sigma : float
+        The estimated noise standard deviation (see section 4.2 of [1]_).
+
+    References
+    ----------
+    .. [1] D. L. Donoho and I. M. Johnstone. "Ideal spatial adaptation
+       by wavelet shrinkage." Biometrika 81.3 (1994): 425-455.
+    """
+    # consider regions with detail coefficients exactly zero to be masked out
+    detail_coeffs = detail_coeffs[np.nonzero(detail_coeffs)]
+
+    if distribution.lower() == 'gaussian':
+        # 75th quantile of the underlying, symmetric noise distribution:
+        # denom = scipy.stats.norm.ppf(0.75)
+        denom = 0.67448975019608171
+        sigma = np.median(np.abs(detail_coeffs)) / denom
+    else:
+        raise ValueError("Only Gaussian noise estimation is currently "
+                         "supported")
+    return sigma
+
+
+def estimate_sigma(im, multichannel, average_sigmas=True):
+    """
+    Robust wavelet-based estimator of the noise standard deviation.
+
+    Parameters
+    ----------
+    im : ndarray
+        Image for which to estimate the noise standard deviation.
+    multichannel : bool
+        Estimate sigma separately for each channel.
+    average_sigmas : bool, optional
+        If true, average the channel estimates of `sigma`.  Otherwise return
+        a list of sigmas corresponding to each channel.
+
+    Returns
+    -------
+    sigma : float or list
+        Estimated noise standard deviation(s).  If `multichannel` is True and
+        `average_sigmas` is False, a separate noise estimate for each channel
+        is returned.  Otherwise, the average of the individual channel
+        estimates is returned.
+
+    Notes
+    -----
+    The estimation algorithm is based on the median absolute deviation of the
+    wavelet detail coefficients as described in section 4.2 of [1]_.
+
+    References
+    ----------
+    .. [1] D. L. Donoho and I. M. Johnstone. "Ideal spatial adaptation
+       by wavelet shrinkage." Biometrika 81.3 (1994): 425-455.
+
+    Examples
+    --------
+    >>> import skimage.data
+    >>> from skimage import img_as_float
+    >>> img = img_as_float(skimage.data.camera())
+    >>> sigma = 0.1
+    >>> img = img + sigma * np.random.standard_normal(img.shape)
+    >>> sigma_hat = estimate_sigma(img, multichannel=False)
+    """
+    if multichannel:
```

Comment:
@sciunto.  How would you propose checking?  I don't think we should assume multichannel means RGB or RGBA.  For example, data from hyperspectral imaging could have different number of channels.  I would prefer to err on the side of user control as long as there is a valid use case (even if that means the user could potentially shoot themselves in the foot!).